### PR TITLE
Add Exult Studio icon and desktop launcher file for Linux

### PR DESCRIPTION
**Pull request by fpiesche**
Icon edit by myself using a [public domain image](https://commons.wikimedia.org/wiki/Category:Blueprints#/media/File:(Elevations_of)_1st_and_2nd_floors,_south,_east,_west,_north_(elevations),_details_of_frame,_door,_and_hall_to_closet_(GGUSC-BlackerR-5353).jpg). Icon has been pngcrushed.

I'm afraid I'm not familiar enough with automake to update the .am file here to conditionally add the Studio files when configured with `--enable-exult-studio`...

### CODE REVIEW

**Comment by DominusExult** [comment]
Thanks, it really makes sense to have a desktop launcher and I like the look of your icon, but we already have a certain look for Exult Studio.
In win32/exultstudio.ico
exultstudio

and in macosx/exult_studio.icns
exult_studio

**Comment by fpiesche** [comment]
> Thanks, it really makes sense to have a desktop launcher and I like the look of your icon, but we already have a certain look for Exult Studio. In win32/exultstudio.ico exultstudio
> 
> and in macosx/exult_studio.icns exult_studio

Ahh! I'll adjust the icon there then...

**Comment by Dragon-Baroque** [comment]
Another nitpick : 
- The executable name of Exult Studio in Linux is `exult_studio`, not `exult-studio`,
- To align with macOS, Windows, etc, the two files might better be stored into `linux/` ( or `unix/` if they work also for some of FreeBSD, OmniOS ) than into `desktop/`.

**Comment by wench** [comment]
desktop seems to be the appropriate place for this as that directory already exists for the exult.desktop fileThat directory can already be considered as a platform specific directory for desktop environments that use .desktop files

**Comment by Dragon-Baroque** [comment]
> desktop seems to be the appropriate place for this as that directory already exists for the exult.desktop file. That directory can already be considered as a platform specific directory for desktop environments that use .desktop files.

You are right. Apologies.

**Comment by fpiesche** [comment]
I placed them next to the existing `.desktop` launcher for Exult as that's largely what they'd go with - feasibly the `desktop` directory might just want renaming to `linux`, or maybe `freedesktop` (being the project that sets the standards for things like `.desktop` files and the AppStream metadata files here)?

Also, is there by any chance a higher-resolution version of the Exult/Studio icons around? The largest size I can see in the .icns files is 128x128, and a common recommendation now for Linux desktops is 256x256 (or just straight up an SVG vector if available)

**Comment by wench** [comment]
If there isn't a higher res version in the source tree, one probably doesn't exist. I doubt anyone even remembers who made it. There are higher res versions of the silver ankh and it  be possible to recreate the map and ruler to make a higher res version of the icon.  The map is obviously a screenshot of exult running in map editor mode. As is windows and macros icons are not completely identical so Linux could have a new icon made with the same style.

**Comment by Dragon-Baroque** [comment]
> I'm afraid I'm not familiar enough with automake to update the .am file here to conditionally add the Studio files when configured with --enable-exult-studio...

This should work, could you try it, replace, in `desktop/Makefile.am`
```
icondir = $(ICONDIR)
dist_icon_DATA = exult.png

desktopdir = $(DESKTOPDIR)
dist_desktop_DATA = exult.desktop
```
by
```
icondir = $(ICONDIR)
desktopdir = $(DESKTOPDIR)

if BUILD_STUDIO
dist_desktop_DATA = exult.desktop exult-studio.desktop
dist_icon_DATA = exult.png exult-studio.png
else
dist_desktop_DATA = exult.desktop
dist_icon_DATA = exult.png
endif
```

**Comment by DominusExult** [comment]
Let me see after the weekend. When I made the macOS icon I created a higher resolution one. Just had to make it a tad different as the macOS style allows more than the usual square

**Comment by DominusExult** [comment]
exultstudioicon_nice
Exult_ankh_nice
exultstudioicon_background
that's 256x256 - thanks to modern technology being able to resize nicely :)

**Comment by fpiesche** [comment]
I'll do you one better - I'm nearly finished redrawing both the ankh and the Studio map background as SVG vector images, for arbitrary rotation and scaling in the future :) These upscales have been very useful in figuring out the detail of the icons! Will update this PR to include the SVGs and 256x icons exported from them once I'm done...

**Comment by wench** [comment]
I could supply the 3d model of the ankh if you wanted but I wouldn't all that useful the requirements of an svg are quite different to a 3d model. I just made it from a tracing of the ankh that came with Ultima9.

**Comment by DominusExult** [comment]
I wouldn‘t mind having svg, icons, 3D models ;)
Updating our icons throughout the different platforms is a good idea.
Many thanks for this PR, which started a fruitful discussion ;)

**Comment by fpiesche** [comment]
> I could supply the 3d model of the ankh if you wanted but I think it wouldn't all that useful as the requirements of an svg are quite different to a 3d model. I just made it from a tracing of the ankh that came with Ultima9.

Yeah, a 3D model wouldn't have been super useful I think. BUT, here's a sneak peek of my vector redraw next to the Studio icon in Inkscape ;)
image

**Comment by fpiesche** [comment]
IT IS DONE

So now here's a question: I've made two versions of the Studio icon - one that's faithful to the original in its composition, just with rounded corners and slightly reduced in size to fit into the [FreeDesktop "reasonable footprint" template](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/quality-guidelines/#reasonable-footprint), and one that's built for a slightly more interesting silhouette. Which one would you prefer? I'll be happy to upload the svgs and 512x pngs for both of them, just trying to figure out which one to pick for the `.desktop` shortcut :)

exult-studio-oldstyle
exult-studio-newstyle

**Comment by marzojr** [comment]
Both are very nice; I *slightly* prefer the top one, but by very little. The main thing I don't like in the bottom image is the tilted ruler; it feels off, although I can't articulate *why*. Can you do a mixed version with the map-ankh composition of the bottom one, but with the ruler-ankh composition of the top one?

**Comment by DominusExult** [comment]
So basically the one I did for macOS (just better looking ;))

**Comment by fpiesche** [comment]
My main goal with the angled ruler was to break the grid on the map up a bit visually, as well as balancing out the ankh poking over the edges of the map a bit. How's this? Or is the ruler better entirely within the boundaries of the map?
info exult exult studio original-256x

File: desktop/info.exult.exult.svg
Comment (fpiesche) [inline desktop/info.exult.exult.svg:1]: Well that's a pain in the neck. Investigating what the deal is with safely exporting mesh gradients from Inkscape...

**Comment by marzojr** [comment]
> My main goal with the angled ruler was to break the grid on the map up a bit visually, as well as balancing out the ankh poking over the edges of the map a bit.

I think that this is part of what was throwing me off the old one: the existence of the grid to begin with. *Because* the grid is there, the ruler being on a "random" orientation throws the icon off-balance, at least for my OCD brain.

> How's this? Or is the ruler better entirely within the boundaries of the map? (the footprint grid overlay is just for reference and won't be in the final icon obviously)
 This looks a LOT better, and it is my favorite of the 3 versions so far. I am curious, though: can you move the ruler right until its top left corner is exactly at the footprint circle (or maybe slightly inside it)?

**Comment by Dragon-Baroque** [comment]
Would you consider naming the Studio desktop files and icons as
`info.exult.studio.desktop` instead of `info.exult.exult.studio.desktop`
and similar ?

Or `info.exult.ExultStudio.desktop` facing `info.exult.Exult.desktop` and similar ?

**Comment by fpiesche** [comment]
> Would you consider naming the Studio desktop files and icons as `info.exult.studio.desktop` instead of `info.exult.exult.studio.desktop` and similar ?
> 
> Or `info.exult.ExultStudio.desktop` facing `info.exult.Exult.desktop` and similar ?

Done! I've refrained from using capitals as that's discouraged by the FreeDesktop spec as there are systems (crucially, flathub.org as the main public repository for Flatpak distibutions) that don't support this. For the Flatpak distribution, due to similar Flatpak-specific restrictions, I'll need to rename the Studio `.desktop` file back to `info.exult.exult.studio` but as that'll be contained entirely within the Flatpak package that shouldn't be an issue.

Also, here's the vector icon with the ruler shuffled to inside the circle footprint and I honestly like that better than the previous revision with it being all the way at the the edge of the icon! I've given it a very slight drop shadow to separate it from the background a little more as well, though a less pronounced one than the ankh.
info exult studio

File: desktop/info.exult.exult.svg
Comment (fpiesche) [reply desktop/info.exult.exult.svg:1]: Looks like the problem with the svg is that while Inkscape supports gradient meshes and can edit and render them fine, they're an in-progress part of the SVG 2.0 spec and not generally supported outside of Inkscape. With that in mind I'll revise the svg to make the vector curves up there a little more complex and use multiple regular gradients, so hat desktops that support svg icons can just use them directly.

**Comment by marzojr** [comment]
> Also, here's the vector icon with the ruler shuffled to inside the circle footprint and I honestly like that better than the previous revision with it being all the way at the the edge of the icon! I've given it a very slight drop shadow to separate it from the background a little more as well, though a less pronounced one than the ankh.

This last version definitely gets my vote; I had a feeling that moving it right would look better, and **love** the drop shadow for the ruler.

**Review by marzojr — APPROVED** [review]

**Review by DominusExult — APPROVED** [review]

**Comment by fpiesche** [comment]
There we go! Redrew the mesh gradients on the SVG ankh so it would work properly with SVG 1.1 compliant renderers and updated both the SVGs and the exported PNGs from it - this means the SVG can both be used as an arbitrarily-scalable icon/logo for e.g. web content or documentation, as well as directly as an icon for Linux desktops supporting SVG icons :)

**Review by marzojr — APPROVED** [review]

**Comment by DominusExult** [comment]
This looks great! Thank you for putting so much work into this. 

Squash and merge, create a merge commit or rebase and merge? :)

I will also update the Exult and Exult Studio icons with this.

Btw. does Windows allow such icons (elements "escaping" the icon boundaries)?

**Comment by fpiesche** [comment]
> Squash and merge, create a merge commit or rebase and merge? :)
I'm fine with squash and merge!

> I will also update the Exult and Exult Studio icons with this.
If you prefer I could export a set of resolutions from the svg directly to a multi-size .ico (and probably .icns for macOS too) and include those here?

> Btw. does Windows allow such icons (elements "escaping" the icon boundaries)?
Yup - Windows XP introduced the ability to have 32bit-colour icons with 8bit alpha channels for transparent backgrounds at up to 128x128 (I think?), and by now Microsoft's .ico format has been extended to also allow much larger resolutions as high-resolution screens are becoming increasingly common.

**Comment by DominusExult** [comment]
> > I will also update the Exult and Exult Studio icons with this.
> 
> If you prefer I could export a set of resolutions from the svg directly to a multi-size .ico (and probably .icns for macOS too) and include those here?
> 
> > Btw. does Windows allow such icons (elements "escaping" the icon boundaries)?
> 
> Yup - Windows XP introduced the ability to have 32bit-colour icons with 8bit alpha channels for transparent backgrounds at up to 128x128 (I think?), and by now Microsoft's .ico format has been extended to also allow much larger resolutions as high-resolution screens are becoming increasingly common.

I wouldn't mind if you were to export those yourself :) 
Best to do it in another PR so if anyone has a different opinion on the Windows icon, it's less messy. Thanks a lot!

**Pull request merged**